### PR TITLE
Blog: Portswigger lab: Broken MFA

### PR DESCRIPTION
**Pull request by psiinon**
Submitted on behalf of @jan-ambroz

Ready for review but to be published on April 6th.

<details>
<summary>Original screenshot</summary>

Screenshot 2022-04-05 at 15-15-55 PortSwigger Labs 2FA Broken Logic

</details>

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+- blog
+- fuzzer
+- portswigger-lab
+date: "2022-04-06"
+authors: 
+    - jan
+---
+
+### The PortSwigger WebSecurity Academy
+
```
Comment (kingthorin) [inline site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md:18]:
```suggestion
```

Drop a blank line.

File: site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+The [PortSwigger WebSecurity Academy](https://portswigger.net/web-security) is a great way to learn more about web security, 
+but not too surprisingly it is also intended as a showcase for Burp Suite.
+
+However you can of course use ZAP instead of Burp, and in this series of ZAP Blog posts I will show you how.
+
+The second lab has the tag PRACTITIONER and you can find it [here](https://portswigger.net/web-security/authentication/multi-factor/lab-2fa-broken-logic). 
+I want to show you how to solve this lab using ZAP and specifically the [Fuzzer](/docs/desktop/addons/fuzzer/) tool.
+
+### Lab instructions
```
Comment (kingthorin) [inline site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md:28]:
```suggestion
### Lab Instructions
```

File: site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+* Your credentials: wiener:peter
+* Victim's username: carlos
+
+You also have access to the email server to receive your 2FA verification code.
+
+### Manual Testing
+
+After logging into our account and using verification code, we closely explore __POST /login2__ request and we can see that the `verify` parameter is used to determine which user's account is being accessed.
+
```
Comment (kingthorin) [inline site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md:40]:
```suggestion
```

File: site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+![Login Requests](images/login-requests.png)
+
+If we try to login to this account multiple times, we see that the verification code is always a 4 digit number. 
+ 
+Log out from your account. 
+
+### Attacking with ZAP
+
+1. First we need to resend the __GET:login2__ request to force the system to generate verification code for Carlos.
```
Comment (kingthorin) [inline site/content/blog/2022-04-06-portswigger-lab-2fa-broken-logic/index.md:52]: "the" or "a" ..... "verification code for..."

**Comment by psiinon** [comment]
Updated to use Numberzz and Expand:

Screenshot 2022-04-06 at 12-34-02 PortSwigger Labs 2FA Broken Logic

Ready for review again and can be published once approved.

**Review by kingthorin — APPROVED** [review]

**Comment by psiinon** [comment]
Ugh, took the screenshot before fixing Rick's comments, new one: 

Screenshot 2022-04-06 at 12-42-52 PortSwigger Labs 2FA Broken Logic

**Comment by kingthorin** [comment]
Thanks

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you both!

**Pull request merged**